### PR TITLE
#171913621 setup development environment

### PR DESCRIPTION
**Pull request by Habinezajanvier**
#### What does this PR do?
Setup development environment 
#### Description of Task to be completed?
* Set up mongoose
* set up express
* set up mongodb
`https://www.digitalocean.com/community/tutorials/how-to-integrate-mongodb-with-your-node-application`
#### How should this be manually tested?
* clone the repo and checkout the branch `ch-setup-development-enviroment-#171913621`
* After checking out the branch, then,
* Copy the content of `.env` file and paste it in your `.env` file.
```javascript
    //.env file   copy data belew and edit where it is necessary for your .env file
USER = yourDbUsername
PASS = yourDbPassword
HOST = 127.0.0.1
PORT = 27017
DB_NAME = yourDbName
```
* Create node environment by running 
```
$export NODE_ENV="development" 
```
* Start the server by running `npm run dev`
* In the command prompt you should see **port** our app is running on and **Database name** it is connected on

*in Postman* 
> On `localhost:3000/lostAndFound/lost` send a *Post* request of simple data like
```json
{
    "name":"Habineza",
    "number":"0789121324
}
```
> You should see a successful message or unsuccessful message
#### What are the relevant pivotal tracker stories?
#171913621

### CODE REVIEW

File: src/db/connection.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import mongoose from 'mongoose';
+import dotenv from 'dotenv';
+dotenv.config();
+
+export const db_connections =()=>{
+    const db_user = process.env.USER;
+    const db_password = process.env.PWD;
+    const db_host = process.env.HOSTNAME;
+    const db_port = process.env.PORTNAME;
```
Comment (YvesIraguha) [inline src/db/connection.js]:
```suggestion
    const db_port = process.env.PORT_NAME;
```

File: src/db/connection.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import mongoose from 'mongoose';
+import dotenv from 'dotenv';
+dotenv.config();
+
+export const db_connections =()=>{
+    const db_user = process.env.USER;
+    const db_password = process.env.PWD;
+    const db_host = process.env.HOSTNAME;
```
Comment (YvesIraguha) [inline src/db/connection.js]: I would like to suggest that you use dash between two words in the same variable

```suggestion
    const db_host = process.env.HOST_NAME;
```

**Review by YvesIraguha — CHANGES_REQUESTED** [review]
Hello @Habinezajanvier thanks for the initiative you had for setting up MongoDB, however, on the pull request description, there are no clear steps for setting up MongoDB. It would be nice if you have added a description or a link to a video on how we can set up MongoDB. In addition, ensure that there is a configuration for `development, production, and testing` environments in database connections as both of these environments are crucial for the success of this project. 

Lastly, remove every single content you have in the `controllers` directory, as we do not need it for now. Even models, it would be better to remove them.

**Review by SimonImanigirimpuhwe — APPROVED** [review]
After reviewing the request by running the app in my local machine I found it working. Therefore, it can be merged.

File: src/db/connection.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        { useNewUrlParser: true,
+            useFindAndModify: false,
+            useCreateIndex:true,
+            useUnifiedTopology: true  },
+        (err) => {
+            if(err){
+                console.log('Failed to connect to MongoDB');
+            }else{
+                console.log(`Connected to ${name} database`);
+            }
```
Comment (YvesIraguha) [inline src/db/connection.js]: @Habinezajanvier can you remove these `console.log`?

**Review by YvesIraguha — CHANGES_REQUESTED** [review]

**Review by YvesIraguha — APPROVED** [review]

**Pull request merged**